src/btree/bt_delete.c

```diff
@@ -293,19 +293,31 @@ __wt_delete_page_rollback(WT_SESSION_IMPL *session, WT_REF *ref)
  *     IDs in the delete info. Since we're called at the end of recovery there's no need to lock the
  *     ref or worry about races.
  */
-static void
+static int
 __delete_redo_window_cleanup_internal(WT_SESSION_IMPL *session, WT_REF *ref)
 {
     WT_REF *child;
 
     WT_ASSERT(session, F_ISSET(ref, WT_REF_FLAG_INTERNAL));
     if (ref->page != NULL) {
         WT_INTL_FOREACH_BEGIN (session, ref->page, child) {
-            if (child->state == WT_REF_DELETED && child->page_del != NULL)
+            /*
+             * We used to do the cleanup when the state of the child was WT_REF_DELETED, but it can
+             * also be loaded in memory if the page has been read before, thus we no longer need
+             * this check anymore.
+             */
```

Comment:
I think it isn't helpful to talk about code that is no longer there and not needed. Someone a year from now will not get new information from that. How about something like:
"Cleanup any page delete structures in the child. We cannot know its state because it could have been loaded in memory if the page was read before this."

Reply:
@sueloverso, I agree. Would you be ok with the following suggestion:
```
Cleanup any page delete structures in the child. At this stage, it can contain stale data we should not rely on.
```

Reply:
Sounds good to me

Reply:
Updated in f162fe8db63a92854e5d86b2e94ac6003125356d